contracts/Governance/Timelock.sol

```diff
@@ -1,123 +1,231 @@
+        TransactionStatus status;
+    }
+
+    uint256 public minimumExecutionDelay;
+    uint256 public maximumExecutionDelay;
+    uint256 public executionGracePeriod;
+
+    address public proposer;
+    EnumerableSet.AddressSet private executors;
+
+    EnumerableSet.Bytes32Set private pendingTransactionsParamsHashes;
+    mapping(bytes32 => TransactionStatus) public pendingTransactions;
+
+    event QueuedTransactionsBatch(bytes32 indexed txParamsHash, uint256 numberOfTransactions, uint256 eta);
+    event CancelledTransactionsBatch(bytes32 indexed txParamsHash);
+    event ApprovedTransactionsBatch(bytes32 indexed txParamsHash);
+    event ExecutedTransaction(bytes32 indexed txParamsHash, address indexed recipient, uint256 value, bytes data, uint256 eta);
+    event NewProposerSet(address indexed previousProposer, address indexed newProposer);
+    event NewExecutorAdded(address indexed executor);
+    event ExecutorRemoved(address indexed executor);
+    event NewExecutionDelaySet(uint256 indexed delay);
+    event NewMinimumExecutionDelaySet(uint256 indexed delay);
+    event NewMaximumExecutionDelaySet(uint256 indexed delay);
+    event NewExecutionGracePeriodSet(uint256 indexed gracePeriod);
+
+    constructor(
+        address _proposer,
+        address _executor,
+        uint256 _minimumExecutionDelay,
+        uint256 _maximumExecutionDelay,
+        uint256 _executionGracePeriod
+    ) {
+        require(_minimumExecutionDelay <= _maximumExecutionDelay, "Timelock: The minimum delay cannot be larger than the maximum execution delay");
+
+        setMinimumExecutionDelay(_minimumExecutionDelay);
+        setMaximumExecutionDelay(_maximumExecutionDelay);
+        setExecutionGracePeriod(_executionGracePeriod);
+        setProposer(_proposer);
+        addExecutor(_executor);
+    }
+
+    function setProposer(address _proposer) public onlyOwner {
+        require(_proposer != address(0), "Timelock: Proposer address cannot be 0");
+        require(_proposer != proposer, "Timelock: New proposer must be different than the current proposer");
+
+        address previousProposer = proposer;
+        proposer = _proposer;
+
+        emit NewProposerSet(previousProposer, proposer);
+    }
+
+    function addExecutor(address _executor) public onlyOwner {
+        require(!executors.contains(_executor), "Timelock: executor already exists");
+        require(_executor != address(0), "Timelock: executor cannot be 0 address");
+
+        executors.add(_executor);
+        emit NewExecutorAdded(_executor);
+    }
+
+    function removeExecutor(address _executor) public onlyOwner {
+        require(executors.contains(_executor), "Timelock: executor doesn't exist");
+
+        executors.remove(_executor);
+        emit ExecutorRemoved(_executor);
+    }
+
+    function executorsCount() external view returns (uint256) {
+        return executors.length();
+    }
+
+    function executorAt(uint256 i) external view returns (address) {
```

Comment:
Just making sure with you - unlike arrays that has that by just being `public`, in this case it doesn't work and you really need to expose it?

Reply:
I'm not user if I understand the question, but I'll try to answer :D Enumerable sets **must be private** (or the compiler complains) so there's no other way to expose neither the sets nor their members in any other way I can think of.